tests/testthat/test-rng.R

```diff
@@ -133,15 +143,79 @@ test_that("poisson numbers", {
 
 
 test_that("Big poisson numbers", {
-  m <- 100000
+  n <- 100000
   lambda <- 20
-  ll <- rep(lambda, m)
-  ans1 <- test_rng_pois(ll, 1L, 1L)
-  ans2 <- test_rng_pois(ll, 1L, 1L)
-  ans3 <- test_rng_pois(ll, 2L, 1L)
+
+  ans1 <- dust_rng$new(1, 1)$rpois(n, lambda)
+  ans2 <- dust_rng$new(1, 1)$rpois(n, lambda)
+  ans3 <- dust_rng$new(2, 1)$rpois(n, lambda)
   expect_identical(ans1, ans2)
   expect_false(all(ans1 == ans3))
 
   expect_equal(mean(ans1), lambda, 1e-2)
   expect_equal(var(ans1), lambda, 1e-2)
 })
+
+
+test_that("norm_rand agrees with rnorm", {
+  n <- 100000
+  ans <- dust_rng$new(2, 1)$norm_rand(n)
+  expect_equal(mean(ans), 0, tolerance = 1e-2)
+  expect_equal(var(ans), 1, tolerance = 1e-2)
+  expect_gt(ks.test(ans, "pnorm")$p.value, 0.1)
+})
+
+
+test_that("continue stream", {
+  rng1 <- dust_rng$new(1, 1L)
+  rng2 <- dust_rng$new(1, 1L)
+
+  y1 <- rng1$runif(100, 0, 1)
+  y2_1 <- rng2$runif(50, 0, 1)
+  y2_2 <- rng2$runif(50, 0, 1)
+  y2 <- c(y2_1, y2_2)
+  expect_identical(y1, y2)
+})
+
+
+test_that("recycle failure", {
+  p <- runif(10)
+  expect_equal(recycle(p[[1]], 10), rep(p[[1]], 10))
+  expect_equal(recycle(p, 10), p)
+  expect_error(recycle(p, 5),
+               "Invalid length for 'p', expected 1 or 5")
+  expect_error(recycle(p, 15),
+               "Invalid length for 'p', expected 1 or 15")
+})
+
+
+test_that("jump", {
+  seed <- 1
+  rng1a <- dust_rng$new(seed, 1L)
+  rng1b <- dust_rng$new(seed, 1L)$jump()
+  rng2 <- dust_rng$new(seed, 2L)
+
+  r2 <- rng2$unif_rand(20)
+  r1a <- rng1a$unif_rand(10)
+  r1b <- rng1b$unif_rand(10)
+
+  m1 <- rbind(r1a, r1b, deparse.level = 0)
+  m2 <- matrix(r2, 2)
+  expect_equal(m1, m2)
+})
+
+
+test_that("long jump", {
+  seed <- 1
+  rng1 <- dust_rng$new(seed, 1L)
```

Comment:
I think the ideal test would be to check that the following are different:
```suggestion
  rng1 <- dust_rng$new(seed, 1L)$jump()
  rng2 <- dust_rng$new(seed, 1L)$long_jump()$jump()
```
As the use of `long_jump()` is so that you can call `jump()` on other processes, without needing to know how many jumps have been called

Reply:
ok cool - I added it as a 4th combination and test all comparisons of the 4

Reply:
(9028408)